File: LeetCode/cpp/2506_Count_Pairs_Of_Similar_Strings.cpp
class Solution {
public:
	int similarPairs(vector<string>& words) {
		int answer = 0;
		vector<string> updated_words;
		unordered_map<string, int> count;

		for (int i = 0; i < words.size(); i++) {
			map<char, int> um;

			for (int j = 0; j < words[i].size(); j++) {
				um[words[i][j]]++;
			}

			string temp;
			for (auto m : um) {
				temp += m.first;
			}

			updated_words.push_back(temp);
		}

		for (int i = 0; i < updated_words.size() - 1; i++) {
			for (int j = i + 1; j < updated_words.size(); j++) {
				if (updated_words[i] == updated_words[j]) {
					answer++;
				}
			}
		}

		return answer;
	}
};

File: LeetCode/java/2506_Count_Pairs_Of_Similar_Strings.java
import java.util.HashMap;
import java.util.Map;
import java.util.TreeMap;

class Solution {
    public int similarPairs(String[] words) {
        int answer = 0;
        String[] updatedWords = new String[words.length];
        Map<String, Integer> count = new HashMap<>();

        for (int i = 0; i < words.length; i++) {
            Map<Character, Integer> map = new TreeMap<>();

            for (int j = 0; j < words[i].length(); j++) {
                map.put(words[i].charAt(j), map.getOrDefault(words[i].charAt(j), 0) + 1);
            }

            StringBuilder temp = new StringBuilder();
            for (Character key : map.keySet()) {
                temp.append(key);
            }

            updatedWords[i] = temp.toString();
        }

        for (int i = 0; i < updatedWords.length - 1; i++) {
            for (int j = i + 1; j < updatedWords.length; j++) {
                if (updatedWords[i].equals(updatedWords[j])) {
                    answer++;
                }
            }
        }

        return answer;
    }
}